Source/UncompressedVideoSampleProvider.cpp

```diff
@@ -132,6 +132,7 @@ IMediaStreamDescriptor UncompressedVideoSampleProvider::CreateStreamDescriptor()
 
     if (codecPar->color_primaries != AVCOL_PRI_UNSPECIFIED && (codecPar->color_primaries < MFVideoPrimaries_BT2020 || applyHdrColorInfo))
     {
+        hasHdrMetadata = true;
```

Comment:
This is actually not correct. If the color_primaries are < BT2020, this is not HDR but SDR. Also for SDR we can have color primaries. This is why we have an or condition with applyHdrColorInfo.